BackEnd/src/main/kotlin/cub/trackmyoffer/FeatureProviderRouting.kt

```diff
@@ -256,7 +256,27 @@ fun Route.featureProviderRouting(httpClient: HttpClient, config: FeatureProvider
                 parameter("notes", notes)
             }
             call.respondText(response.bodyAsText(), status = response.status)
-        }    }
+        }
+
+        post("/analyze-gaps") {
+            val extractorResponse = extractJobDescription(call)
+            if (extractorResponse.status != HttpStatusCode.OK) {
+                call.respond(extractorResponse.status, extractorResponse.body)
+                return@post
+            }
+
+            val profileId = userIdSupplier()
```

Comment:
We have this issue in multiple endpoints, but I believe we should try to extract the user id before doing any calls to fp in `extractJobDescription`, even though it doesn't expose anything personal